src/utils/DateUtils.ts

```diff
@@ -29,111 +130,174 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
+    const today = moment(DateUtils.keepOnlyDatePart(options.now));
     if (options.useTodayYesterdayAndTomorrow) {
-      if (dateOnly.valueOf() == today.valueOf()) {
-        return l('Today');
-      }
-      var tomorrow = DateUtils.offsetDateByDays(today, 1);
-      if (dateOnly.valueOf() == tomorrow.valueOf()) {
-        return l('Tomorrow');
-      }
-      var yesterday = DateUtils.offsetDateByDays(today, -1);
-      if (dateOnly.valueOf() == yesterday.valueOf()) {
-        return l('Yesterday');
+      /*
+       'today' is already a moment, but we need to create another instance of moment so that 'today' isn't modified by 'add'.
+       If we simply used 'today', we would have to add(-2,'days') instead of -1.
+       */
+      if (dateOnly.valueOf() == today.valueOf() || dateOnly.valueOf() == moment(today).add(1, 'days').valueOf() ||
+        dateOnly.valueOf() == moment(today).add(-1, 'days').valueOf()) {
+        return moment(dateOnly).calendar(moment(today));
       }
     }
 
-    var isThisWeek = Math.abs(TimeSpan.fromDates(dateOnly, today).getDays()) < 7;
+
+    const isThisWeek = dateOnly.diff(moment(today), 'weeks') == 0;
     if (options.useWeekdayIfThisWeek && isThisWeek) {
       if (dateOnly.valueOf() > today.valueOf()) {
-        return l('NextDay', Globalize.format(dateOnly, 'dddd'));
+
+        return l('Next') + ' ' + dateOnly.format('dddd');
+      } else if (dateOnly.valueOf() < today.valueOf()) {
+        return l('Last') + ' ' + dateOnly.format('dddd');
       } else {
-        return l('LastDay', Globalize.format(dateOnly, 'dddd'));
+        return dateOnly.format('dddd');
       }
     }
 
-    if (options.omitYearIfCurrentOne && dateOnly.getFullYear() === today.getFullYear()) {
-      return Globalize.format(dateOnly, 'M');
+    if (options.omitYearIfCurrentOne && dateOnly.year() === today.year()) {
+      return dateOnly.format('MMMM DD');
     }
 
     if (options.useLongDateFormat) {
-      return Globalize.format(dateOnly, 'D');
+      return dateOnly.format('dddd, MMMM DD, YYYY');
     }
 
-    return Globalize.format(dateOnly, 'd');
+    return dateOnly.format('M/D/YYYY');
   }
 
+  /**
+   * Converts the time information from a date into a string.
+   * @param date The date containing the time information to format.
+   * @returns {any} Will either be an empty string or a string containing the time.
+   */
   static timeToString(date: Date, options?: IDateToStringOptions): string {
     if (Utils.isNullOrUndefined(date)) {
       return '';
     }
 
-    return Globalize.format(date, 't');
+    return moment(date).format('h:mm A');
   }
 
+  /**
+   * Checks if options and their time conditions are met, and returns the date accordingly using timeToString.
+   * @param date The date to format according to the specified options.
+   * @param options The options specified to format the date according to preference.
+   * @returns {any} The date after being formatted according to the specified options.
```

Comment:
I would add comments explaining the (subtle) difference between this method, `dateToString` and `dateTimeToString`.

And I would do the same explanation in each methods.